docs/adr/0004-instrumentation.md

```diff
@@ -0,0 +1,100 @@
+takes over 8 hours to complete. We intend to reduce GLAM ETL time by finding the
+probes that are most likely to be requested by users from the aforementioned
+data and have the ETL build only those probes. This document also presents a
+plan for ad hoc probe building, which needs to happen when someone requests a
+probe that had not been pre-built by the ETL.
+
+## Decision Drivers
+
+- Usage metrics collection must be reliable and cannot be bypassed or influenced
+  by other systems such as browsers' ad blockers
+- GLAM ETL should consume this service so it can optimize its work, only
+  processing probes that are likely to be looked at
+- GLAM App should consume this service so it can trigger an ad hoc calculation
+  whenever a probe that was not processed by GLAM ETL is requested
+- Ad hoc probe building should be independent of GLAM App because it's likely to
+  be CPU/memory intensive, with different needs in terms of scalability
+
+## Considered Options
+
+- 1: Collect metrics on the front end (via a library such as Open Telemetry
+  Instrumentation or a custom-built solution) and send them to the GLAM back end
+  server to be stored in a table in the GLAM App Postgres DB. Expose meaningful
+  api endpoints for usage data consumption by GLAM App and GLAM ETL. This
+  provides flexibility to adapt to future requirements and "immunity" to
+  adblockers, but might be longer to implement
+- 2: Use Gleanjs, a JS library for telemetry, which would be quicker than a
+  custom implementation. It's also good because of dogfooding and internal
+  support. It can be bypassed by adblockers, but we can work around the issue by
+  having metrics be sent to a custom endpoint in Glam's domain.
+- 3: Google Analytics: currently available, it's quick and easy to implement,
+  but doesn't give us the insight into e.g. session-scoped data. We can make it
+  resilient against adblockers by sending metrics to our own endpoint, but in
+  order to do that we'd have to implemet Google's Measurement Protocol in our
+  backend so it can process metrics properly, and even if we achieve that we'd
+  still want the current GA to keep working as it is, which involves sending
+  metrics to Google's endpoint hence making GA vulnerable to adblockers. One way
+  out of this is capture all Glam metrics from GA through a custom endpoint, but
+  then this starts to grow way beyond the scope of this work.
+- 4: Option #1, but instead of using GLAM's Postgres DB, save metrics to a
+  BigQuery table. This would make it easy for the ETL to consume the data, but
+  will most likely slow down some GLAM App operations due to BigQuery's
+  slower-than-OLTP response times
+- 5: Stackdriver + gcloud monitoring tools: This would provide an interface for
+  exporting metrics and the ability to create dashboards for visualizing them,
+  but since our use case also involves consumption of those metrics by the GLAM
```

Comment:
I don't think consuming probe usage metrics for dynamic ETL is a hard requirement. If you have a set of most important probes and only ETL those, shouldn't the ad-hoc probe builder should handle the rest?

I think the probe scraper dag in WTMO our most failure prone.